lsp/core/aws-lsp-core/src/content/schemaProviderBuilder.test.ts

```diff
@@ -0,0 +1,81 @@
+import * as chai from 'chai'
+import { expect } from 'chai'
+import * as chaiAsPromised from 'chai-as-promised'
+import { beforeEach, describe, it } from 'node:test'
+import { URI } from 'vscode-uri'
+import {
+    ContentRequestMiddleware,
+    ContentRequestMiddlewareDelegate,
+    ContentRequestResponse,
+} from './contentRequestMiddleware'
+import { SchemaProviderBuilder } from './schemaProviderBuilder'
+
+chai.use(chaiAsPromised)
+
+describe('Test SchemaProviderBuilder', async () => {
+    const sampleUri = 'file://sample/location'
+    let sut: SchemaProviderBuilder
+
+    beforeEach(async () => {
+        sut = new SchemaProviderBuilder()
+    })
+
+    it('throws when no handler is added', async () => {
+        const handler = sut.build()
+        await expect(handler(sampleUri)).to.be.rejected
+    })
+
+    it('throws when unhandled', async () => {
+        sut.addHandler(createDelegatingMiddleware())
+
+        const handler = sut.build()
+        await expect(handler(sampleUri)).to.be.rejected
+    })
+
+    it('builds one handler', async () => {
+        sut.addHandler(createMiddlewareReturning('foo'))
+
+        const handler = sut.build()
+        const result = await handler(sampleUri)
+
+        expect(result).to.equal('foo')
+    })
+
+    it('delegates to next handler', async () => {
+        sut.addHandler(createDelegatingMiddleware())
+        sut.addHandler(createMiddlewareReturning('foo'))
```

Comment:
Minor:  The tricky thing with this test is if the order were to be reversed by accident in SchemaProviderBuilder then the last handler returning 'foo' would always be called and returned first which will pass the test.  Is there a way for this test to ensure that the other handler was in fact called?

Reply:
Yes, good catch.